lib/common/zstd_internal.h

```diff
@@ -137,7 +137,10 @@ static const U32 OF_defaultNormLog = 5;
 /*-*******************************************
 *  Shared functions to include for inlining
 *********************************************/
-static void ZSTD_copy8(void* dst, const void* src) { memcpy(dst, src, 8); }
+static void ZSTD_copy8(void* dst, const void* src)
+{
+    MEM_write64(dst, MEM_read64(src));
```

Comment:
I did not identify a specific case.  I just felt that if someone has disabled the `memcpy()` version, then it shouldn't be used here either.

If you are concerned about the load/store potentially being suboptimal then one option is to only do it when `memcpy()` is disabled:

```
static void ZSTD_copy8(void* dst, const void* src)
{
#if defined(MEM_FORCE_MEMORY_ACCESS) && (MEM_FORCE_MEMORY_ACCESS>=1)
    MEM_write64(dst, MEM_read64(src));
#else
    memcpy(dst, src, 8);
#endif
}
```

I would also consider moving the "copy" functions into `mem.h`.